ouroboros-consensus-cardano/src/ouroboros-consensus-cardano/Ouroboros/Consensus/Cardano/CanHardFork.hs

```diff
@@ -712,3 +718,17 @@ translateValidatedTxBabbageToConwayWrapper ::
        (ShelleyBlock (Praos c) (ConwayEra c))
 translateValidatedTxBabbageToConwayWrapper ctxt = InjectValidatedTx $
     fmap unComp . eitherToMaybe . runExcept . SL.translateEra ctxt . Comp
+
+{-------------------------------------------------------------------------------
+  Auxiliary
+-------------------------------------------------------------------------------}
+
+-- TODO abstract this out more
+untickShelleyLedgerState ::
+     TickedLedgerState (ShelleyBlock proto era)
+  -> LedgerState       (ShelleyBlock proto era)
+untickShelleyLedgerState st = ShelleyLedgerState {
```

Comment:
Would you pattern match on `st` in a way that makes it locally obvious that this is half of an isomorphism?

Reply:
Hmm, will this always be true? In general, I think ticked ledger states could contain strictly more information than unticked ones.

Or are you only saying that this is true **ATM**, and it would be neat to locally highlight that?

Reply:
Yeah, the second one.